src/python/WMCore/MicroService/MSPileup/MSPileupTasks.py

```diff
@@ -100,17 +105,131 @@ def pileupSizeTask(self):
+        - create new rules for custom DID via provided rseExpression
+        - set expiration date (to be 24h) for already existing ruleIds from pileup document
+        - add new ruleIds to pileup document
+
+        :return: None
+        """
+        spec = {'active': True}
+        docs = self.mgr.getPileup(spec)
+        for doc in docs:
+            self.logger.info("partialPileupTask process %s", doc)
+            fraction = doc['containerFraction']
+
+            # check previous fraction value and proceed only if has changed
+            previousFraction = 1.0
+            transition = doc.get('transition', [])
+            if transition:
+                previousFraction = transition[-1]['containerFraction']
+            if previousFraction == fraction:
+                # do nothing since there is no transition, i.e. skip this record
+                self.logger.info("Pileup name %s has no container fractions changes", doc['pileupName'])
+                continue
+
+            # get rucio DIDs (datasets, in CMS name blocks) for our pileup document as following:
+            # - when container fraction is decreasing: new container is a subset of customName
+            # - when container fraction is increasing: new container is a superset of customName + subset of pileupName
+
+            # usage of block names defined in this logic:
+            # https://github.com/dmwm/WMCore/pull/11807#pullrequestreview-1786778783
+            pname = doc['pileupName']
+            blockNames = self.rucioClient.getBlocksInContainer(pname)
+            if previousFraction < fraction:
+                # if it increasing
+                # use EVERY single block defined in the custom dataset plus blocks from the standard pileup
+                cname = doc.get('customName', '')
+                if cname:
+                    blockNames = self.rucioClient.getBlocksInContainer(cname) + blockNames
+            else:
+                # if it is decreasing
+                # instead of using random blocks from the standard pileup dataset (pileupName)
+                # we should use it from the custom dataset (customName) - if existent
+                cname = doc.get('customName', '')
+                if cname:
+                    blockNames = self.rucioClient.getBlocksInContainer(cname)
+
+            # get portion of DIDs based on ceil(containerFraction * num_rucio_datasets)
+            portion = blockNames[:math.ceil(fraction * len(blockNames))]
+            self.logger.info("For pileup document %s take portion %s", doc, portion)
+
+            # create new container DID in Rucio for our custom name
+            self.rucioClient.createContainer(doc['customName'], scope=self.customRucioScope)
+
+            # we call attachDIDs Rucio wrapper API with our set of DIDs
+            newRules = []
+            for rse in doc['currentRSEs']:
+                self.rucioClient.attachDIDs(rse, doc['customName'], portion, scope=self.customRucioScope)
```

Comment:
This process deserves attention and potentially a discussion with DM experts. Here are some of the concerns I have:
a) can we attach the same DID but against a different RSE?
b) I know the Rucio wrapper does not accept a list at the moment. But does Rucio server accept it?
c) depending on increasing/decreasing, we will likely have to call this attachment twice. i) for blocks that we know their current location/RSE; ii) for blocks that are potentially nowhere on Disk (hence, rse=None).

Reply:
apart from your comment it is unclear what exactly you expect me to put in a code except saying we should be aware and discuss it further. Said that, I put your comment inside the code to address it later. If you want to address this issue in this PR please be concrete and specify which steps out of above list we should do.

Reply:
According to Stefano Belforte, he/CRAB creates custom container and he does not provide any RSEs when attaching DIDs to a container. Based on that, I would suggest to pass `None` as RSE name (in practice, not setting any RSE when attaching a Rucio dataset to a Rucio container).

I still don't know how in practice that is going to be reflected in Rucio server, and what latency it's going to cause. But I believe it to be the safest approach at the moment.